src/main/java/at/hannibal2/skyhanni/features/rift/everywhere/PunchcardHighlight.kt

```diff
@@ -0,0 +1,76 @@
+package at.hannibal2.skyhanni.features.rift.everywhere
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.HypixelData
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.IslandChangeEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.RenderMobColoredEvent
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.EntityUtils.isNPC
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import net.minecraft.client.Minecraft
+import net.minecraft.client.entity.EntityPlayerSP
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class PunchcardHighlight {
+    private val config get() = SkyHanniMod.feature.rift.punchcard
+    private val ownIGN get() = Minecraft.getMinecraft().thePlayer.name
+    private var lastRiftServer: String = ""
+
+    private val patternGroup = RepoPattern.group("rift.punchcard")
+    private val punchedPattern by patternGroup.pattern(
+        "new",
+        "§5§lPUNCHCARD! §r§eYou punched §r§[\\da-z](?:.*?)?(\\w+)§r§[\\da-z] §r§eand both regained §r§a\\+25ф Rift Time§r§e!"
+    )
+    private val repeatPattern by patternGroup.pattern(
+        "repeat",
+        ""
+    )
+
+    private val playerList: MutableSet<String>?
+        get() = ProfileStorageData.profileSpecific?.rift?.punchedPlayers
```

Comment:
Why is this in the Storage. As soon as you switch lobby in the rift the counter resets. The only benefit of storage is if you close minecraft and then rejoin to the same rift lobby we would remember which people you have punched.

Reply:
well yeah that unique use case is the only reason it's here, but if it's that unique of a usecase i can just remove it